dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala

```diff
@@ -182,11 +182,11 @@ object EnceladusRestDAO extends EnceladusDAO {
 
         if (isUnauthorized) {
           log.warn(s"Unauthorized GET request for Menas URL: $url")
+          if (retriesLeft == 0) {
+            throw UnauthorizedException(s"Unable to reauthenticate after retries")
+          }
           log.warn(s"Expired session, reauthenticating")
           if (enceladusLogin()) {
```

Comment:
if the login fails won't it go straight to else on line 193 which will throw an exception that will be caught by line 210 and there will be no retries? Meaning never more than 1 retrie for logging in.

Reply:
It's intended to only have 1 retry.